iree/turbine/kernel/wave/templates/speculative_decoding.py

```diff
@@ -420,18 +438,19 @@ def inner_loop(
                 tkw.set_symbol(DRAFT_TOKEN_ID, draft_token_id)
                 tkw.set_symbol(CUR_PROB_OFFSET, cur_prob_offset)
                 target_prob_single = read_3d_into_1d(target_probs)
+                prob_acc += target_prob_single
 
                 condition = (coin <= (prob_acc / threshold_acc_reg)) | (
                     target_prob_single >= threshold_single_reg
                 )
                 not_condition = ~condition
 
-                # Update num_accepted_tokens if the condition is true.
-                num_accepted_tokens = tkw.select(
-                    condition, num_accepted_tokens + one, num_accepted_tokens
+                # Update prob_acc.
```

Comment:
Seems suspicious. Is this double addition also in the CUDA kernel?

Reply:
this should be select(cond, 0, prob_acc) otherwise you are adding target_prob_single twice.